### PR TITLE
feat: limit x-axis range

### PR DESCRIPTION
**Pull request by rephus**
Closes:  #2943

### Description:

This issue https://github.com/apache/echarts/issues/16984#issuecomment-1849116907 is still affecting these charts. 

Screenshot from 2023-12-11 11-41-14

Screenshot from 2023-12-11 11-41-38

Screenshot from 2023-12-11 11-41-48

Screenshot from 2023-12-11 12-03-04

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by rephus** [comment]
## Known limitation

When a chart is flipped (and the xAxis becomes the yAxis) the min/max range doesn't work, 

This is because when the axis changes, it goes from `type: value` to `type: category`

This should be fixed in a separate PR 

Screenshot from 2023-12-11 12-08-48
Screenshot from 2023-12-11 12-09-12

**Comment by ZeRego** [comment]
Label isn't correct. Should say `x` and `bottom` 

Screenshot 2023-12-12 at 09 42 14

Range input didn't show when I used order date (day)

Screenshot 2023-12-12 at 09 41 32

https://lightdash-pr-8261.onrender.com/share/34jDN2xe1xKe1UP-MtHVj

File: packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts [hunk 1]
```diff
@@ -991,14 +990,12 @@ const getEchartAxis = ({
...
-                    : undefined,
-                max: validCartesianConfig.layout.flipAxes
-                    ? xAxisConfiguration?.[1]?.max ||
-                      maybeGetAxisDefaultMaxValue(allowSecondAxisDefaultRange)
-                    : undefined,
+                min:
+                    xAxisConfiguration?.[1]?.min ||
+                    maybeGetAxisDefaultMinValue(allowSecondAxisDefaultRange),
+                max:
+                    xAxisConfiguration?.[1]?.min ||
```
Comment (ZeRego) [inline packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts:997]:
```suggestion
                    xAxisConfiguration?.[1]?.max ||
```

Comment (rephus) [reply packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts]: good catch, thanks

**Comment by rephus** [comment]
> Label isn't correct. Should say `x` and `bottom`

Fixed, thanks

Screenshot from 2023-12-12 13-50-12

Screenshot from 2023-12-12 13-50-03

Also fixed dates

**Review by ZeRego — APPROVED** [review]

**Pull request merged**